src/kafka_plugin.c

```diff
@@ -194,6 +194,48 @@ void kafka_plugin(int pipe_fd, struct configuration *cfgptr, void *ptr)
   }
 #endif
 
+  /* setting some defaults */
+  if (!config.sql_host)
+    config.sql_host = default_kafka_broker_host;
+  if (!config.kafka_broker_port)
+    config.kafka_broker_port = default_kafka_broker_port;
+
+  if (!config.sql_table)
+    config.sql_table = default_kafka_topic;
+
+#if defined(WITH_AVRO) && defined(WITH_SERDES)
+  if (config.kafka_avro_schema_registry) {
+    if (strchr(config.sql_table, '$')) {
+      Log(LOG_ERR, "ERROR ( %s/%s ): dynamic 'kafka_topic' is not compatible with 'avro_schema_registry'. Exiting.\n",
+          config.name, config.type);
+      exit_gracefully(1);
+    }
+
+    p_kafka_init_host_struct(&kafkap_kafka_host);
```

Comment:
Hi Uwe ( @ustorbeck ),

Thanks for this very useful contribution.

The whole patch does makes sense to me; the only thing that leaves me puzzled is using p_kafka_init_host_struct() inside the if section: should the structure not be initialized regardless? Exactly like p_kafka_init_host_conf() is unconditionally executed in kafka_cache_purge()?

Paolo